zio-crypto/shared/src/main/scala/zio/crypto/random/SecureRandom.scala

```diff
@@ -0,0 +1,107 @@
+package zio.crypto.random
+
+import zio._
+import zio.crypto.ByteHelpers
+
+import java.security.{ NoSuchAlgorithmException, SecureRandom => JSecureRandom }
+
+object SecureRandom {
+
+  type SecureRandom = Has[SecureRandom.Service]
+
+  trait Service {
+    def nextBytes(length: Int): Task[Array[Byte]]
+    def nextString(entropyBytes: Int): Task[String]
+    def setSeed(seed: Long): UIO[Unit]
+    def getJavaSecureRandom: UIO[JSecureRandom]
```

Comment:
Another pattern that libraries have used:

```scala
def execute[A](f: JSecureRandom => A): ZIO[Any, Throwable, A]
```

This way it's harder to "leak" the mutable interface.